File: SampleApp/Objective-C/SampleApp/VideoInterstitial/VideoInterstitialViewController.m
//
//  VideoInterstitialViewController.m
//  SampleApp
//
//  Fluct SDK
//  Copyright © 2020 fluct, Inc. All rights reserved.
//

#import "VideoInterstitialViewController.h"
@import FluctSDK;

static NSString *const kVideoInterstitialGroupID = @"1000104107";
static NSString *const kVideoInterstitialUnitID = @"1000160561";

@interface VideoInterstitialViewController () <FSSVideoInterstitialDelegate>
@property (weak, nonatomic) IBOutlet UIButton *showButton;
@property (nonatomic) FSSVideoInterstitial *interstitial;
@end

@implementation VideoInterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    FSSVideoInterstitialSetting *settings = FSSVideoInterstitialSetting.defaultSetting;
    settings.debugMode = YES;
    self.interstitial = [[FSSVideoInterstitial alloc] initWithGroupId:kVideoInterstitialGroupID
                                                               unitId:kVideoInterstitialUnitID
                                                              setting:settings];
    self.interstitial.delegate = self;
}

- (IBAction)didTouchUpLoadAd:(id)sender {
    [self.interstitial loadAd];
}

- (IBAction)didTouchUpShowAd:(id)sender {
    if ([self.interstitial hasAdAvailable]) {
        [self.interstitial presentAdFromViewController:self];
    }
}

#pragma mark - FSSVideoInterstitialDelegate

- (void)videoInterstitialDidLoad:(FSSVideoInterstitial *)interstitial {
    NSLog(@"video interstitial ad did load");
    self.showButton.enabled = YES;
}

- (void)videoInterstitial:(FSSVideoInterstitial *)interstitial didFailToLoadWithError:(NSError *)error {
    NSLog(@"video interstitial ad load failed. Because %@", error);
}

- (void)videoInterstitialWillAppear:(FSSVideoInterstitial *)interstitial {
    NSLog(@"video interstitial ad will appear");
}

- (void)videoInterstitialDidAppear:(FSSVideoInterstitial *)interstitial {
    NSLog(@"video interstitial ad did appear");
}

- (void)videoInterstitial:(FSSVideoInterstitial *)interstitial didFailToPlayWithError:(NSError *)error {
    NSLog(@"video interstitial ad play failed. Because %@", error);
}

- (void)videoInterstitialWillDisappear:(FSSVideoInterstitial *)interstitial {
    NSLog(@"video interstitial ad will disappear");
    self.showButton.enabled = NO;
}

- (void)videoInterstitialDidDisappear:(FSSVideoInterstitial *)interstitial {
    NSLog(@"video interstitial ad did disappear");
}

@end

File: SampleApp/Swift/SampleApp/VideoInterstitial/VideoInterstitialViewController.swift
//
//  VideoInterstitialViewController.swift
//  SampleApp
//
//  Fluct SDK
//  Copyright © 2020 fluct, Inc. All rights reserved.
//

import UIKit
import FluctSDK

// グループID, ユニットIDを設定して下さい
private let fluctVideoInterstitialGroupID = "1000104107"
private let fluctVideoInterstitialUnitID = "1000160561"

class VideoInterstitialViewController: UIViewController {

    @IBOutlet weak var showButton: UIButton?

    private var interstitial: FSSVideoInterstitial?

    override func viewDidLoad() {
        super.viewDidLoad()

        let setting = FSSVideoInterstitialSetting.default
        setting.isDebugMode = true
        self.interstitial = FSSVideoInterstitial(groupId: fluctVideoInterstitialGroupID,
                                                 unitId: fluctVideoInterstitialUnitID,
                                                 setting: setting)
        self.interstitial?.delegate = self
    }

    @IBAction func didTouchUpLoadAd(_ sender: Any) {
        self.interstitial?.loadAd()
    }

    @IBAction func didTouchUpShowAd(_ sender: Any) {
        guard let interstitial = self.interstitial, interstitial.hasAdAvailable() else { return }
        interstitial.presentAd(from: self)
    }

}

extension VideoInterstitialViewController: FSSVideoInterstitialDelegate {

    func videoInterstitialDidLoad(_ interstitial: FSSVideoInterstitial) {
        print("video interstitial ad \(interstitial) did load")
        showButton?.isEnabled = true
    }

    func videoInterstitial(_ interstitial: FSSVideoInterstitial, didFailToLoadWithError error: Error) {
        print("video interstitial ad \(interstitial) load failed. Because \(error.localizedDescription)")
    }

    func videoInterstitialWillAppear(_ interstitial: FSSVideoInterstitial) {
        print("video interstitial ad \(interstitial) will appear")
    }

    func videoInterstitialDidAppear(_ interstitial: FSSVideoInterstitial) {
        print("video interstitial ad \(interstitial) did appear")
    }

    func videoInterstitial(_ interstitial: FSSVideoInterstitial, didFailToPlayWithError error: Error) {
        print("video interstitial ad \(interstitial) play failed. Because \(error.localizedDescription)")
    }

    func videoInterstitialWillDisappear(_ interstitial: FSSVideoInterstitial) {
        print("video interstitial ad \(interstitial) will disappear")
    }

    func videoInterstitialDidDisappear(_ interstitial: FSSVideoInterstitial) {
        print("video interstitial ad \(interstitial) did disappear")
        showButton?.isEnabled = false
    }

}
